source/slang/slang-emit-wgsl.cpp

```diff
@@ -0,0 +1,968 @@
+
+    m_writer->emit("fn ");
+    m_writer->emit(name);
+
+    emitSimpleFuncParamsImpl(func);
+
+    // An absence of return type is expressed by skipping the optional '->' part of the
+    // header.
+    if (resultType->getOp() != kIROp_VoidType)
+    {
+        m_writer->emit(" -> ");
+        emitType(resultType);
+    }
+}
+
+void WGSLSourceEmitter::emitSimpleFuncParamImpl(IRParam* param)
+{
+    if (auto sysSemanticDecor = param->findDecoration<IRTargetSystemValueDecoration>())
+    {
+        m_writer->emit("@builtin(");
+        m_writer->emit(sysSemanticDecor->getSemantic());
+        m_writer->emit(")");
+    }
+
+    CLikeSourceEmitter::emitSimpleFuncParamImpl(param);
+}
+
+void WGSLSourceEmitter::emitMatrixType(
+    IRType *const elementType, const IRIntegerValue& rowCountWGSL,
+    const IRIntegerValue& colCountWGSL
+    )
+{
+    // WGSL uses CxR convention
+    m_writer->emit("mat");
+    m_writer->emit(colCountWGSL);
+    m_writer->emit("x");
+    m_writer->emit(rowCountWGSL);
+    m_writer->emit("<");
+    emitType(elementType);
+    m_writer->emit(">");
+}
+
+void WGSLSourceEmitter::emitStructDeclarationSeparatorImpl()
+{
+    m_writer->emit(",");
+}
+
+static bool isPowerOf2(const uint32_t n)
+{
+    return (n != 0U) && ((n - 1U) & n) == 0U;
+}
+
+void WGSLSourceEmitter::emitStructFieldAttributes(
+    IRStructType * structType, IRStructField * field
+    )
+{
+    // Tint emits errors unless we explicitly spell out the layout in some cases, so emit
+    // offset and align attribtues for all fields.
+    IRSizeAndAlignmentDecoration *const sizeAndAlignmentDecoration =
+        structType->findDecoration<IRSizeAndAlignmentDecoration>();
+    // NullDifferential struct doesn't have size and alignment decoration
+    if (sizeAndAlignmentDecoration == nullptr)
+        return;
+    SLANG_ASSERT(sizeAndAlignmentDecoration->getAlignment() > IRIntegerValue{0});
+    SLANG_ASSERT(
+        sizeAndAlignmentDecoration->getAlignment() <= IRIntegerValue{UINT32_MAX}
+    );
+    const uint32_t structAlignment =
+        static_cast<uint32_t>(sizeAndAlignmentDecoration->getAlignment());
+    IROffsetDecoration *const fieldOffsetDecoration =
+        field->findDecoration<IROffsetDecoration>();
+    SLANG_ASSERT(fieldOffsetDecoration->getOffset() >= IRIntegerValue{0});
+    SLANG_ASSERT(fieldOffsetDecoration->getOffset() <= IRIntegerValue{UINT32_MAX});
+    SLANG_ASSERT(isPowerOf2(structAlignment));
+    const uint32_t fieldOffset =
+        static_cast<uint32_t>(fieldOffsetDecoration->getOffset());
+    // Alignment is GCD(fieldOffset, structAlignment)
+    // TODO: Use builtin/intrinsic (e.g. __builtin_ffs)
+    uint32_t fieldAlignment = 1U;
+    while (((fieldAlignment & (structAlignment | fieldOffset)) == 0U))
+        fieldAlignment = fieldAlignment << 1U;
+
+    m_writer->emit("@align(");
+    m_writer->emit(fieldAlignment);
+    m_writer->emit(")");
+}
+
+bool WGSLSourceEmitter::isPointerSyntaxRequiredImpl(IRInst* inst)
```

Comment:
I don't think you need this anymore by making `doesTypeSupportPointers` to `false`.